setuptools/glob.py

```diff
@@ -0,0 +1,169 @@
+"""
+Filename globbing utility. Mostly a copy of `glob` from Python 3.5.
+
+Changes include:
```

Comment:
It would not be possible to ship the glob library from Py3.5 without any modifications, as `yield from` is not valid syntax in older Pythons, so at least that needs to change regardless.

To make the library compatible with Py2, I also had to use `six` for dealing with strings/bytes. These changes require modifications to internal function code, so are not easily patched from "outside". I notice that I did not mention this in the list of changes, I will add it now.

The same problem exists with including hidden files - `is_hidden` is called from a number of functions, so these functions need to be changed to not call it. Alternatively, `is_hidden` could be changed to always return `False`, which would have the same effect.

Finally, all the functions in the glob library call each other directly. There is no good way of doing dependency injection or something similar to override the specific functions that need patching.

I completely agree with you that this adds some undesirable technical debt. If you can think of any good solutions to the problems above I am very happy to implement them.